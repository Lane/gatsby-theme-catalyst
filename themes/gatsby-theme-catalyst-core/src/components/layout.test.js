import React from "react"
import renderer from "react-test-renderer"
import SiteLayout from "./layout"
import { useStaticQuery, StaticQuery } from 'gatsby';
import catalystConfig from "../../../../jest/__mocks__/site-config"

describe("Header", () => {
  const props = {
    children: 'test',
    ...catalystConfig,
  }



  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }) => (
        render(props)
      ),
      useStaticQuery.mockReturnValue(props)
    );
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(<SiteLayout {...props}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})