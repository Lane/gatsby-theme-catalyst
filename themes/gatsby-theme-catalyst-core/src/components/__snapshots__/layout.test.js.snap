// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="css-0"
>
  <div
    className="css-155ux7s-SiteContainer"
  >
    <header
      className="css-92rb9y-SiteHeader"
      id="header"
    >
      <p>
        Header area for latent shadowing in sibling themes
      </p>
    </header>
    <main
      className="css-1spq66s-SiteMain"
    >
      <div
        className="css-1gjvup9-ContentContainer"
      >
        test
      </div>
    </main>
    <footer
      className="css-o72uzv-SiteFooter"
      id="footer"
    >
      <p>
        Footer area for latent shadowing in sibling themes
      </p>
    </footer>
  </div>
</div>
`;
